datascience/tables.py

```diff
@@ -1585,30 +1590,30 @@ def with_column(self, label, values):
         ValueError: Column length mismatch. New column does not have the same number of rows as table.
         """
         new_table = self.copy()
-        new_table.append_column(label, values)
+        new_table.append_column(label, values, formatter)
         return new_table
 
-    def with_columns(self, *labels_and_values):
+    def with_columns(self, *labels_and_values_and_formats):
```

Comment:
I think the right signature is 

    def with_columns(self, *labels_and_values, *, formatter=None)

and it's limited to giving the same formatter to every new column or no formatter at all.